cloudinit/atomic_helper.py

```diff
@@ -34,8 +42,15 @@ def b64e(source):
 def write_file(
     filename, content, mode=_DEF_PERMS, omode="wb", preserve_mode=False
 ):
-    # open filename in mode 'omode', write content, set permissions to 'mode'
+    """open filename in mode omode, write content, set permissions to mode"""
+
+    with performance.Timed(f"Writing {filename}"):
```

Comment:
Can the decorator be used here now instead?

Reply:
The only reason that I didn't is that I couldn't think of a clean way to still include the filename in the log, which I think is useful for this log.